### ISSUE TITLE
Adding Maps through Directory Browser is inconsistent

### ISSUE THREAD

**Issue by axelalex2** [ManiaControl/ManiaControl#257]
If you add Map A + Map B through the Directory Browser to the Map List, Map B will appear first in the Map List, but second in the Map Queue. This is especially annoying in events, where you rotate through the same Map Order every time. You would need to add the Maps in reverse order, so that the order in the Map List is correct, clear the Map Queue, and (if needed) re-queue the Maps through the Map List.
There need to be two Feature Extensions: first, give the ability to either add the new Map at the end of the List or to insert it after the current Map; and second, an option to queue up the new Map as well. To be decided if in the Directory Browser itself, or as a general setting.

**Comment by kremsy** [comment]
I dont understand the problem. If you need s certain order for a event you will add the maps beforehand anyways? Than you easily can sort them or queue them in correct order.

**Comment by axelalex2** [comment]
True, that you have enough time to order the maps before an event, but still, the current implementation with the described issue is not intuitive, especially if you don't recheck on the map list/queue after adding the maps. Also, ordering in ManiaControl itself is not easy, since the map list does not have a reordering feature available.

**Comment by kremsy** [comment]
Yeah agree, our cup plugin works kinda this way:

All not added maps are added without putting them into queue, clear queue, queue in correct order.
